keras/backend/tensorflow_backend.py

```diff
@@ -507,7 +507,12 @@ def relu(x, alpha=0., max_value=None):
     if max_value is not None:
         x = tf.clip_by_value(x, tf.cast(0., dtype=_FLOATX),
                              tf.cast(max_value, dtype=_FLOATX))
-    x -= tf.constant(alpha, dtype=_FLOATX) * negative_part
+    try:
+        alpha = tf.constant(alpha, dtype=_FLOATX)
+    except TypeError:
+        # in case alpha is not a const
+        pass
+    x -= alpha * negative_part
```

Comment:
`try / except` is not the proper thing to do here. You could do a type check instead.